src/fileio/datafile.cxx

```diff
@@ -539,6 +541,8 @@ bool Datafile::write(const char *format, ...) const {
   tmp.openw(filename);
   bool ret = tmp.write();
   tmp.close();
+
+  delete[] filename;
```

Comment:
`filename` shadows `filename` of the class. Therefore the local version needs to be deleted here.
The local version is needed, as this is in a const method.